MetaMorpheus/GUI/Views/CustomProteaseWindow.xaml.cs

```diff
@@ -0,0 +1,127 @@
+            string psiMsAccessionNum = PsiMsAccessionNumTextBox.Text;
+            string psiMsName = PsiNameTextBox.Text;
+            string motifString = DigestionMotifTextBox.Text;
+            Dictionary<string, CleavageSpecificity> CleavageSpecDic = new Dictionary<string, CleavageSpecificity>();
+            foreach (CleavageSpecificity c in Enum.GetValues(typeof(CleavageSpecificity))) {
+                CleavageSpecDic.Add(c.ToString(), c);
+            }
+            CleavageSpecificity cleavageSpecificity = CleavageSpecDic[CleavageSpecComboBox.Text];
+            List<DigestionMotif> digestionMotifs;
+
+            try
+            {
+                digestionMotifs = DigestionMotif.ParseDigestionMotifsFromString(motifString);
+            }
+            catch
+            {
+                MessageBox.Show("The motif(s) could not be parsed.");
+                return;
+            }
+
+            //load the default ProteaseDictionary
+            //string folderPath = Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles);
+            //string path = ((!string.IsNullOrWhiteSpace(folderPath) && AppDomain.CurrentDomain.BaseDirectory.Contains(folderPath) && !AppDomain.CurrentDomain.BaseDirectory.Contains("Jenkins")) ? System.IO.Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "MetaMorpheus") : AppDomain.CurrentDomain.BaseDirectory);
+            //string path2 = System.IO.Path.Combine(path, "ProteolyticDigestion", "proteases.tsv");
+            //ProteaseDictionary.LoadProteaseDictionary(path2);
+
+            //create a new Protease and add to ProteaseDictionary
+            Protease proteaseToAdd = new Protease(proteaseName, cleavageSpecificity, psiMsAccessionNum, psiMsName, digestionMotifs);
+
+            if (ProteaseDictionary.Dictionary.ContainsKey(proteaseName))
+            {
+                MessageBox.Show("This protease already exists");
+            }
+            else
+            {
+                ProteaseDictionary.Dictionary.Add(proteaseName, proteaseToAdd);
+            }
+
+            if (!ProteaseDictionary.Dictionary.TryAdd(proteaseName, proteaseToAdd))
+            {
+                MessageBox.Show("Success! Protease '" + proteaseName + "' has been added to the dictionary.");
+            }
+            
+            //customProtease file path
+            string ProtDirectory = Path.Combine(GlobalVariables.DataDir, @"ProteolyticDigestion");
+            string customProteasePath = Path.Combine(ProtDirectory, @"CustomProtease.tsv");
+
+            //check if the customProtease file exists, create it if not
+            if (!File.Exists(customProteasePath))
+            {
+                File.Create(customProteasePath);
+            }
+
+            string lineToAdd = "\n" + proteaseName + "\t" + motifString + "\t\t\t" + CleavageSpecComboBox.Text + "\t" + psiMsAccessionNum + "\t" + psiMsName + "\t\t\t\t\t";
```

Comment:
what pieces of information are being skipped over with all of the \t\t\t\t and is that information important here?